JOING-58 feat: 추천 시스템 API 구현

hyoom opened the pull request:
## #️⃣ 연관된 이슈

> #JOING-58

<br/>

## 📝 작업 내용

> AI 기반으로 추천해주는 기능을 구현하였습니다.
> 기획안(item)정보를 통해 크리에이터를 추천받고, 크리에이터(creator)정보를 통해 기획안을 추천 받도록 하였습니다.
> AI와의 통신은 WebClient를 이용하여 구현하였고 현재는 아직 비동기가 정상적으로 작동하지 않습니다. 추후 메시지 큐를 도입하여 비동기 처리를 개선할 예정입니다.

<br/>

## 📷 스크린샷(선택)

> 이미지가 있다면 첨부해주세요

<br/>

## 💬 리뷰 요구사항(선택)

> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요
>
> ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요?
>

hyoom committed:
JOING-58 feat: 추천 시스템 예외 처리 구현

hyoom committed:
JOING-58 feat: 추천 시스템 DTO 구현

hyoom committed:
JOING-58 feat: 추천 시스템 기능 구현

jayden-bin commented on src/main/java/com/ktb/joing/domain/recommend/dto/request/CreatorRecommendRequest.java:
```diff
@@ -0,0 +1,29 @@
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+public class CreatorRecommendRequest {
+    private String title;
+    @JsonProperty("item_Category")
```
item_Category -> item_category
바꿔야 할것 같아요 ㅋㅋㅋ

jayden-bin commented on src/main/java/com/ktb/joing/domain/recommend/dto/response/ItemRecommend.java:
```diff
@@ -0,0 +1,19 @@
+@Getter
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+public class ItemRecommend {
+    @JsonProperty("item_id")
+    private Long itemId;
```
이런걸 보면 json serializer가 카멜케이스로 되어있는걸 스네이크케이스로 바꾸고 싶어 하시는것 같은데
차라리 글로벌하게 혹은 api 통신 부분에서만 objecMapper나 jsonSerializer를 스네이크케이스를 사용하도록 바꾸는게 좋을것 같아요~!

hyoom replied:
카멜케이스로 되어있는걸 스네이크케이스 변경하느라 자잘한 실수가 많았는데 팁 감사합니다!
AI api 통신 부분에서만 objecMapper를 사용하도록 수정하였습니다

hyoom committed:
Merge branch 'develop' into feature/JOING-58-ai-recommendation

hyoom committed:
JOING-58 refactor: AI 통신에 objectMapper snake case 설정 추가

- 기존: 카멜케이스로 작성된 필드들을 개별 @JsonProperty로 스네이크케이스 변환
- 변경: ObjectMapper 설정으로 카멜케이스 필드들이 자동으로 스네이크케이스로 변환되도록 개선

hyoom committed:
JOING-58 feat: AI 추천 엔드포인트 변경

hyoom committed:
JOING-58 feat: AI 필드 변경사항 반영 및 임시로 작성했던 로직 삭제
